include/GafferScene/MergeScenes.h

```diff
@@ -0,0 +1,141 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENE_MERGESCENES_H
+#define GAFFERSCENE_MERGESCENES_H
+
+#include "GafferScene/SceneProcessor.h"
+
+namespace GafferScene
+{
+
+class GAFFERSCENE_API MergeScenes : public SceneProcessor
+{
+
+	public :
+
+		MergeScenes( const std::string &name=defaultName<MergeScenes>() );
+		~MergeScenes() override;
+
+		GAFFER_GRAPHCOMPONENT_DECLARE_TYPE( GafferScene::MergeScenes, MergeScenesTypeId, SceneProcessor );
+
+		Gaffer::BoolPlug *transformsPlug();
+		const Gaffer::BoolPlug *transformsPlug() const;
+
+		Gaffer::BoolPlug *attributesPlug();
+		const Gaffer::BoolPlug *attributesPlug() const;
+
+		Gaffer::BoolPlug *objectsPlug();
+		const Gaffer::BoolPlug *objectsPlug() const;
+
+		Gaffer::BoolPlug *globalsPlug();
+		const Gaffer::BoolPlug *globalsPlug() const;
```

Comment:
I think personally I'd prefer `merge*Plug` as it saves a few double takes when following the general logic in affects/hash/compute. It think it's also a bit clearer when working with the node in python if the node itself isn't called something "mergey".

Reply:
Makes sense, but given that we've decided to introduce a mode, I've renamed to `*ModePlug` in b8c001a67e88d3fdd957e878ef00c984590e4a0a. This fits with our mode plugs on other nodes.